labs/lab06.md

```diff
@@ -0,0 +1,75 @@
+# Lab 6: Refactoring and Anti-Patterns
+
+## Introduction 
+A lot of the design principles and heuristics we have been encouraging in this class are there to encourage good design *up-front*. But you do not always have control over how a system you are using gets created. 
+
+
+**Anti-patterns** are common examples of bad design. They tend to come about due to a program’s natural evolution, coupled with poor choices early on, which leads to code that is increasingly *tangled*, badly coupled, low on cohesion, etc. These are often signaled by **code smells** -- heuristic, but not guaranteed indicators of anti-patterns, like overly long methods, or a class that calls a lot of another class’ code (feature envy, signals high coupling).
+
+In this lab, you will become a code-smelling agent who detects  and attempt at refactoring them. 
+## Deliverables
+
+- [ ] Identify the anti-pattern associated with Frogger crossing the road and fix it. Use the Appendix for anti-pattern vocabulary!
+- [ ] Identify the anti-pattern associated with Frogger records and fix it.
+- [ ] Identify at least two issues (no need to name the anti-pattern) associated with the Drawing System, and explain how you would refactor it. No coding required.
+
+## Instructions
+
+### Setup
+<u>Fork</u> and clone the repo from [https://github.com/CMU-17-214/f23-lab06](https://github.com/CMU-17-214/f23-lab06.git).
+
+The first two tasks are explicitly referring to Java code (the typescript codebase is also provided, but as longer legacy version from previous semesters). The last task is provided both in Java and Typescript, so feel free to explore the one you are most comfortable with. If you would like additional exercise, feel free to look into the "accounts" folder.
```

Comment:
they are all provided in both languages, right? why not let students pick the language?